spec/aria-reflection.html

```diff
@@ -0,0 +1,761 @@
+            <code>ariaPlaceholder</code> with only the P capitalized.</li>
+          <li>There are currently no acronym-based ARIA attributes, but if future attributes include acronym usage, attempt to match
+            existing DOM conventions (e.g. ID becomes Id).</li>
+        </ul>
+      </section>
+      <section class="informative" id="idl_attr_exceptions">
+        <h3>IDL Attribute Name Notes or Exceptions</h3>
+        <p>Any notes or exceptions for specific attribute names will be listed here.</p>
+        <ul>
+          <li>
+            <code>ariaPosInSet</code>: The
+            <pref>aria-posinset</pref> attribute refers to an item's position in a set (two words: "in set") rather than the "inset"
+            of an item from the beginning of the collection. Therefore the IDL attribute name is
+            <code>ariaPosInSet</code> with the P, I, and second S capitalized,
+            <em>not</em>
+            <code>ariaPosInset</code>.</li>
+        </ul>
+        <p class="ednote">Editor's Note: Should we make an exception on the spelling of "placeholder" and capitalize the H anyway? Some developers
+          will expect this to be
+          <code>ariaPlaceHolder</code> despite the fact that it's not a hyphenated word.</p>
+      </section>
+    </section>
+
+    <section class="informative" id="idl_example_usage">
+      <h2>Example IDL Attribute Usage</h2>
+      <p>The primary purpose of ARIA IDL attribute reflection is to ease JavaScript-based manipulation of values. The following
+        examples demonstrate its usage.</p>
+      <aside class="example">
+
+        <!-- ReSpec needs these examples to be unindented. -->
+        <pre>&lt;div id="inaccessibleButton"&gt;
+    &lt;!-- Use semantic markup instead. This is just a retrofit example. --&gt;
+&lt;/div&gt;</pre>
+        <pre>// Get a reference to the element.
+let el = document.getElementById('inaccessibleButton'); 
+el.tabIndex = 0; // Make it focusable.
+
+// Set the role and label.
+el.role = "button";
+el.ariaLabel = "Edit";
+
+// Get the role and label.
+el.role; // Returns "button"
+el.ariaLabel; // Returns "Edit"
+
+// These are interchangeable with the more verbose setAttribute and getAttribute methods.
+el.setAttribute("role", "button");
+el.setAttribute("aria-label", "Edit");
+el.getAttribute("role"); // Returns "button"
+el.getAttribute("aria-label"); // Returns "Edit"
+
+// Changes via either interface are reflected by the other.
+el.setAttribute("aria-label", "Delete");
+el.ariaLabel; // Returns "Delete"
+el.ariaLabel = "Publish";
+el.getAttribute("aria-label"); // Returns "Publish"</pre>
+
+      </aside>
+    </section>
+
+    <section class="normative" id="shadowroot_interaction">
+      <h2>Interaction between
+        <code>ShadowRoot</code> and
+        <code>host</code>
+      </h2>
+      <p>If an
+        <code>Element</code> has an attached
+        <code>ShadowRoot</code>, and the author has access to the
```

Comment:
Saying "if the author has access" is confusing to me. *Some* author always has access to the ShadowRoot.

Reply:
Not necessarily - if you attach a ShadowRoot and don't keep a reference to it, you will not have access to it.